src/main/java/com/example/onlinebookstore/repository/BookRepository.java

```diff
@@ -3,8 +3,9 @@
 import com.example.onlinebookstore.model.Book;
 import java.util.List;
 import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface BookRepository {
+public interface BookRepository extends JpaRepository<Book, Long> {
```

Comment:
I don't fully understand why you need a declaration of the save, findAll and findById methods if JPA already provides an implementation of them? You can just use them in the controller.